Description: Fixes for iso_c_bind.F95 under gfortran
Author: Alastair McKinstry <mckinstry@debian.org>
Forwarded: no
Last-Updated: 2018-11-29

Index: flang-20180921/runtime/flang/iso_c_bind.F95
===================================================================
--- flang-20180921.orig/runtime/flang/iso_c_bind.F95
+++ flang-20180921/runtime/flang/iso_c_bind.F95
@@ -156,7 +156,7 @@
 #if defined TARGET_LINUX_X8664 || defined TARGET_LLVM_64
         ! added this to make the data-initializd module common block be at 32 bytes long
         ! maybe only needed for linux86-64
-        integer*8 :: __iso_bind_private_i = 11, __iso_bind_private_j = 23
+        integer*8 :: iso_bind_private_i = 11, iso_bind_private_j = 23
 #endif
 
         contains
@@ -173,13 +173,13 @@
             end function
 
             logical function compare_eq_cfunptrs(a,b)
-            type(C_FUNPTR) :: a,b
+            type(C_FUNPTR), intent(IN) :: a,b
             compare_eq_cfunptrs = (a%val .eq. b%val)
             return
             end function
 
             logical function compare_ne_cfunptrs(a,b)
-            type(C_FUNPTR) :: a,b
+            type(C_FUNPTR), intent(IN) :: a,b
             compare_ne_cfunptrs = (a%val .ne. b%val)
             return
             end function
